record-and-playback/core/scripts/events/events.rb

```diff
@@ -0,0 +1,74 @@
+# Set encoding to utf-8
+# encoding: UTF-8
+#
+# BigBlueButton open source conferencing system - http://www.bigbluebutton.org/
+#
+# Copyright (c) 2018 BigBlueButton Inc. and by respective authors (see below).
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 3.0 of the License, or (at your option)
+# any later version.
+#
+# BigBlueButton is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
+#
+
+require '../lib/recordandplayback'
+require 'logger'
+require 'trollop'
+require 'yaml'
+
+def keep_events(meeting_id, redis_host, redis_port, events_dir, break_timestamp)
+  BigBlueButton.logger.info("Keeping events for #{meeting_id}")
+  redis = BigBlueButton::RedisWrapper.new(redis_host, redis_port)
+  events_archiver = BigBlueButton::RedisEventsArchiver.new redis
+  events_xml = "#{events_dir}/#{meeting_id}/events.xml"
+  events = events_archiver.store_events(meeting_id, events_xml, break_timestamp)
```

Comment:
Calling the `store_events` function multiple times with the same break timestamp is not safe, because it deletes the events for the segment after it writes the events.xml file.

You must ensure that store_events is only called once per break point.

In the case of a recorded meeting, I suggest running the normal archive script then copy `events.xml` to the events directory. Only call the `events/events.rb` script on non-recorded meetings.

Reply:
Ok. Thanks @kepstin ! I probably won't need to manually remove the events either, right? https://github.com/pedrobmarin/bigbluebutton/blob/keep-meeting-events/record-and-playback/core/scripts/events/events.rb#L66

Reply:
I recommend leaving that manual event removal code. That is supposed to be done after the final recording segment is saved - it removes some extra things in addition to events like meeting metadata and breakout room information.